x/pki/types/message_add_pki_revocation_distribution_point.go

```
@@ -0,0 +1,183 @@
+}
+
+func (msg *MsgAddPkiRevocationDistributionPoint) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgAddPkiRevocationDistributionPoint) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid signer address (%s)", err)
+	}
+
+	err = validator.Validate(msg)
+	if err != nil {
+		return err
+	}
+
+	isDataDigestInTypes := true
+	if msg.DataDigestType != 0 {
+		isDataDigestInTypes = false
+		for _, digestType := range allowedDataDigestTypes {
+			if digestType == msg.DataDigestType {
+				isDataDigestInTypes = true
+
+				break
+			}
+		}
+	}
+
+	if !isDataDigestInTypes {
+		return pkitypes.NewErrInvalidDataDigestType(fmt.Sprintf("invalid DataDigestType: %d. Supported types are: %v", msg.DataDigestType, allowedDataDigestTypes))
+	}
+
+	if msg.RevocationType != allowedRevocationType {
+		return pkitypes.NewErrInvalidRevocationType(fmt.Sprintf("invalid RevocationType: %d. Supported types are: %d", msg.RevocationType, allowedRevocationType))
+	}
+
+	cert, err := x509.DecodeX509Certificate(msg.CrlSignerCertificate)
+	if err != nil {
+		return pkitypes.NewErrInvalidCertificate(err)
+	}
+
+	subjectAsMap := x509.SubjectAsTextToMap(cert.SubjectAsText)
+
+	if *msg.IsPAA {
+		if msg.Pid != 0 {
+			return pkitypes.NewErrNotEmptyPid("Product ID (pid) must be empty for root certificates when isPAA is true")
+		}
+
+		if !cert.IsSelfSigned() {
+			return pkitypes.NewErrRootCertificateIsNotSelfSigned(fmt.Sprintf("CRL Signer Certificate must be self-signed if isPAA is True"))
+		}
+
+		strVid, found := subjectAsMap["Mvid"]
+		if found {
+			vid, err := strconv.ParseInt(strings.Trim(strVid, "0x"), 16, 32)
+			if err != nil {
+				return err
+			}
+
+			if int32(vid) != msg.Vid {
+				return pkitypes.NewErrCRLSignerCertificateVidNotEqualMsgVid("CRL Signer Certificate vid must equal to message vid")
+			}
+		}
```

If there is no VID for a root cert, let's throw some Unsupported Error (that publishing a revocation point for non-VID scoped root certificates is currently not supported).